src/Shelduck.hs

```diff
@@ -123,6 +131,24 @@ checkTopic b t =
   else info (object ["bad_topic" .= showResult b]) >> return False
   where showResult = pack . show
 
+data DefinitionListRun = DefinitionListRun {
+  _assertionCount       :: Integer,
+  _assertionFailedCount :: Integer
+} deriving (Show)
+
+defaultDefinitionListRun = DefinitionListRun 0 0
+
+$(makeLenses ''DefinitionListRun)
+
+runAssertion :: (MonadIO m) => TVar TopicResult -> WebhookRequest -> StateT DefinitionListRun m TimedResponse
+runAssertion t x = do
+  assertionResult <- liftIO $ runReaderT (performRequest x) t
+  assertionCount += 1
+  if assertionResult ^. testResult == Failed
+    then assertionFailedCount += 1
```

Comment:
These aren't actual mutating increments. See: http://www.haskellforall.com/2013/05/program-imperatively-using-haskell.html for more. TLDR; cheating using [lens](https://hackage.haskell.org/package/lens)